battle.go

```diff
@@ -232,3 +259,13 @@ func (t DamageTransaction) BattleLog() string {
 		t.Damage,
 	)
 }
+
+type ItemTransaction struct {
+	Target *Pokemon
+	Item   *Item
+	Move   *Move
+}
+
+func (t ItemTransaction) BattleLog() string {
+	return fmt.Sprintf("%s used on %s.", t.Item.Name, t.Target.GetName())
```

Comment:
Probably not necessary to change now, but this doesn't really provide a whole lot of information. Doesn't describe stuff like how much hp was restored, or if status effects were removed.

Reply:
Not sure how this should be handled. In theory, using an item and its effect are two separate logs. Using a potion and a healing move both produce an effect that heals a Pokemon, which leads me to believe it should be a separate transaction like `HealTransaction` or similar.